### ISSUE TITLE
[bug] package query does not work for me

### ISSUE THREAD

**Issue by freihamer** [conan-io/conan#15780]
### Environment details

* Operating System+version: Ubuntu 20.04
* Compiler+version:
* Conan version: 2.0.17
* Python version: 3.8

### Steps to reproduce

For me, the package query option does not filter results as expected
`conan list -r myrepo -p os=Linux "*"`
Will still list all packages in myrepo, not only those that have Linux as os

**Comment by memsharded** [comment]
Hi @lucabarbiero 

Thanks for your question. 

I am not sure what you mean. This seems to work fine:
```
(conan2_36) λ conan list -r=conancenter -p os=Linux "zlib/1.3:*"        
conancenter                                                             
  zlib                                                                  
    zlib/1.3                                                            
      revisions                                                         
        5c0f3a1a222eebb6bff34980bcd3e024 (2024-01-23 08:39:53 UTC)      
          packages                                                      
            abe5e2b04ea92ce2ee91bc9834317dbe66628206                    
              info                                                      
                settings                                                
                  ...                                 
                  os: Linux                                             
                                 
            b647c43bfefae3f830561ca202b6cfd935b56205                    
              info                                                      
                settings                                                
                  ...                              
                  os: Linux                                             
                                        
$ conan list -r=conancenter -p os=Windows "zlib/1.3:*"      
conancenter                                                             
  zlib                                                                  
    zlib/1.3                                                            
      revisions                                                         
        5c0f3a1a222eebb6bff34980bcd3e024 (2024-01-23 08:39:53 UTC)      
          packages                                                      
            ae9eaf478e918e6470fe64a4d8d4d9552b0b3606                    
              info                                                      
                settings                                                
                  ...                               
                  os: Windows                                           
```

If you are typing ``"*"`` that is not a pattern for binaries, that is a pattern for recipes. It will return all recipes in the repo.
Next Conan versions might actually throw an error for this case.

**Comment by freihamer** [comment]
All right sorry I think my command would then be 
`conan list -r myrepo -p os=Linux "*:*"`

Solved thanks

**Comment by memsharded** [comment]
No prob, thanks for the feedback, it was still useful! 🙂 

I am submitting PR https://github.com/conan-io/conan/pull/15781 that will raise an error message to explain the behavior, it will help other users with the same potential confusion.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/15781 merged, it will be in next 2.2
